pennylane/_qubit_device.py

```diff
@@ -942,7 +942,7 @@ def adjoint_jacobian(self, tape, starting_state=None, use_device_state=False):
                         'the "adjoint" differentiation method'
                     )
             else:
-                if op.name not in ("QubitStateVector", "BasisState"):
+                if op.name not in ("QubitStateVector", "BasisState", "Snapshot"):
```

Comment:
Out of scope, but wondering: would this be an issue with Barrier too? :thinking:

Reply:
Hmm good question, I'm not sure about other (hardware?) qubit devices, but for our simulator Barrier is an unsupported operation so it is expanded away.